packages/cdk/utils/src/replace-units/replace-units.ts

```diff
@@ -0,0 +1,112 @@
+import { Nullable } from '@angular-ru/cdk/typings';
+
+import { isNil } from '../is-nil';
+import { isNotNil } from '../is-not-nil';
+import { UnitsMap } from './types';
+
+const REGEXP_NUMBER: string = `[0-9]+([\.][0-9]+)?`;
+
+export function replaceUnits(text: string, unitsMap: UnitsMap): string {
+    let result: string = text;
+
+    result = replaceMultiple(result, unitsMap);
+    result = replaceSingle(result, unitsMap);
+
+    return result;
+}
+
+function replaceMultiple(value: string, unitsMap: UnitsMap): string {
+    const unitKeys: string = getUnitKeys(unitsMap);
+    const pattern: RegExp = new RegExp(
+        `(^|\\s)((${REGEXP_NUMBER})(${unitKeys}))(\\s+((${REGEXP_NUMBER})(${unitKeys})))+(\\s|$)`
+    );
+
+    let result: string = value;
+    let match: Nullable<string> = null;
+
+    while (pattern.test(result)) {
+        match = pattern.exec(result)?.[0] ?? '';
+        const num: Nullable<number> = convertToNumberMultiple(match, unitsMap);
+        const unitValue: string = isNil(num) ? '' : num.toString();
+        const prepared: string = match.replace(RegExp(/\S.*\S/), unitValue);
+
+        result = result.replace(pattern, prepared);
+    }
+
+    return result;
+}
+
+function convertToNumberMultiple(inputUnit: string, unitsMap: UnitsMap): Nullable<number> {
+    const unit: string = inputUnit.trim();
+    const sum: number = unit
+        .split(' ')
+        .filter((item: string): boolean => item !== '')
```

Comment:
not sure about safety

Reply:
it is safe because of regexp
input text must be correct